solr/core/src/test/org/apache/solr/handler/admin/MetricsHandlerTest.java

```diff
@@ -325,8 +335,123 @@ public void testKeyMetrics() throws Exception {
     handler.handleRequestBody(req(CommonParams.QT, "/admin/metrics", CommonParams.WT, "json",
         MetricsHandler.KEY_PARAM, "solr.jetty:unknown:baz"), resp);
     values = resp.getValues();
-    metrics = (NamedList)values.get("metrics");
+    metrics = (NamedList) values.get("metrics");
     assertEquals(0, metrics.size());
     assertNotNull(values.findRecursive("errors", "solr.jetty:unknown:baz"));
   }
+
+  @Test
+  public void testMetricsUnload() throws Exception {
```

Comment:
We need another test that tests the following scenario:
* instance 1 registers its gauges
* instance 2 is created and registers its gauges
* instance 1 is closed and should unregister its gauges

together with dependent child instances